source/SVInt.cpp

```diff
@@ -476,6 +476,71 @@ SVInt SVInt::lshr(uint32_t amount) const {
     return result;
 }
 
+SVInt SVInt::ashr(const SVInt& rhs) const {
+    if (!signFlag)
+        return lshr(rhs);
+    if (rhs.hasUnknown())
+        return createFillX(bitWidth, signFlag);
+    if (rhs >= bitWidth)
+        return SVInt(bitWidth, *this >= 0 ? 0 : -1, signFlag);
+
+    uint16_t contractedWidth = bitWidth - rhs.getAssertUInt64();
+    SVInt tmp = lshr(rhs);
+    tmp.bitWidth = contractedWidth;
+    return signExtend(tmp, bitWidth);
+}
+
+SVInt SVInt::ashr(uint32_t amount) const {
+    if (!signFlag)
+        return lshr(amount);
+    if (amount == 0)
+        return *this;
+    if (amount >= bitWidth)
+        return SVInt(bitWidth, *this >= 0 ? 0 : -1, signFlag);
+
+    uint16_t contractedWidth = bitWidth - amount;
+    SVInt tmp = lshr(amount);
+    tmp.bitWidth = contractedWidth;
+    return signExtend(tmp, bitWidth);
+}
+
+SVInt SVInt::ambiguousConditionalCombination(const SVInt& rhs) const {
```

Comment:
now that I think about it, it would be hard to make this a helper method in the evaluator, as it has to access private fields of SVInt (val/pVal) -- we could expose those to friends, or this can stay part of SVInt